src/lsm/compaction.zig

```diff
@@ -522,13 +538,15 @@ pub fn CompactionType(
             // so value_next_equal is false, or a new value is hit, which will increment it.
             const source_count = source_index;
             const target_count = target_index;
-
             assert(target_count <= source_count);
-            assert(target_count > 0);
-
             compaction.context.table_info_a.immutable =
                 compaction.context.table_info_a.immutable[source_count..];
 
+            if (target_count == 0) {
```

Comment:
I know we mention we can't have empty tables in schema.zig, just checking that this won't generate them?

Reply:
It wont, for two reasons:

* this hits the same code path as `compaction.context.table_info_a.immutable.len == 0` case, with the `assert(compaction.values_in[0].len == 0)`.
* that case works correctly, it'll hit `if (values_in[0].len == 0 and values_in[1].len == 0)` condition if level b is also empty